src/starkware/starknet/cli/starknet_cli.py

```diff
@@ -106,12 +106,30 @@ async def invoke_or_call(args, command_args, call: bool):
         raise ValueError('Invalid address format.')
     for abi_entry in abi:
         if abi_entry['type'] == 'function' and abi_entry['name'] == args.function:
+            previous_input = None
+            current_inputs_ptr = 0
+            for input_desc in abi_entry['inputs']:
+                assert current_inputs_ptr < len(args.inputs), \
+                        f'Expected at least {current_inputs_ptr + 1}, got {len(args.inputs)}'
+                current_input_val = args.inputs[current_inputs_ptr]
```

Comment:
```
                assert current_inputs_ptr < len(args.inputs), \
                        f'Expected at least {current_inputs_ptr + 1}, got {len(args.inputs)}'
                current_input_val = args.inputs[current_inputs_ptr]
```
The current input value is only interesting when the argument is of type `felt`, thus this part should be moved to the next condition clause. In addition, the assertion may not hold when the argument is of type `felt*` and it is OK (e.g., when the length of the array is 0 and hence there is no need for a `felt*` argument after it)

Reply:
That's very true. It's also incorrect when the invocation doesn't have any arguments. Let me think.

Reply:
Nevermind, it's correct in my case since we won't enter the loop. Thinking about yours.

Reply:
I agree :)